core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -585,6 +602,62 @@ extern "C" fn new_show_error(n_args: usize, args: *const Obj, kwargs: *mut Map)
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_confirm_webauthn(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let icon: Option<StrBuffer> = kwargs.get(Qstr::MP_QSTR_icon)?.try_into_option()?;
+        let app_names: Obj = kwargs.get(Qstr::MP_QSTR_app_names)?;
+        let account_name: Obj = kwargs.get(Qstr::MP_QSTR_account_names)?;
+
+        // Need to find the overall page count for pagination purposes.
+        let page_count = {
+            let mut count = 0;
+            let mut iter_buf = IterBuf::new();
+            let iter = Iter::try_from_obj_with_buf(app_names, &mut iter_buf)?;
+            for _ in iter {
+                count += 1;
+            }
+            count
+        };
+
+        // Closure to lazy-load the information on given page index.
+        // Done like this to allow arbitrarily many pages without
+        // the need of any allocation here in Rust.
+        let get_page = move |page_index| {
+            let app_name: StrBuffer = {
+                let mut iter_buf = IterBuf::new();
+                let iter = unwrap!(Iter::try_from_obj_with_buf(app_names, &mut iter_buf));
+                iter.skip(page_index).next().unwrap().try_into().unwrap()
+            };
+
+            let account_name: Option<StrBuffer> = {
+                let mut iter_buf = IterBuf::new();
+                let iter = unwrap!(Iter::try_from_obj_with_buf(account_name, &mut iter_buf));
+                iter.skip(page_index)
+                    .next()
+                    .unwrap()
+                    .try_into_option()
+                    .unwrap()
+            };
+
+            return FidoPage::new(app_name, account_name);
+        };
+        let pages = FidoPages::new(get_page, page_count, "_tmp".into());
+
+        let controls = Button::cancel_confirm(
+            Button::with_icon(theme::ICON_CANCEL).styled(theme::button_cancel()),
+            Button::with_text("CONFIRM").styled(theme::button_confirm()),
+            2,
+        );
+
+        let fido_page = FidoPaginatedPage::new(pages, icon, controls);
+
+        let obj = LayoutObj::new(Frame::new(title, fido_page))?;
```

Comment:
The right border is too thin, fix with:
```suggestion
        let obj = LayoutObj::new(Frame::new(title, fido_page).with_border(theme::borders()))?;
```

Frame normally expects the inner component to have vertical scrollbar which is not the case here. This is quite ugly but using negative borders seemed uglier to me and couldn't think of anything else.

Reply:
Done in [e47408f](https://github.com/trezor/trezor-firmware/pull/2556/commits/e47408f74d63837b6d43ba0266f49ee74bd3f0e3)

All other places using `with_border` combine it with `into_child`, so I made it as well

Reply:
Yep, it should be everywhere (but isn't). I thought it wouldn't compile with anything else but actually got confused by the trait names. Will add this to my TODO.

Reply:
#2581